app/journey/Journey.scala

```diff
@@ -135,6 +134,10 @@ final case class Journey(
     ddRef                     = ddRef.map(_ => "***"),
     maybeSaUtr                = maybeSaUtr.map(_ => "***")
   )
+
+  override def toString: String = {
```

Comment:
In order to be consistent there you have to either:
1. Change `toString` methods in other case classes participating in definition of a journey
2. or use a `obfuscate` method whenever you print the journey.

Now it's a mixture of both.

Reply:
I'd lean a little towards the `toString` methods being overriden. 
Reason being that the PII could get leaked in the messages of surprise exceptions which could get thrown outside of our code (e.g. json deserialisation) and they'll be using the `toString` on these.

Reply:
My feeling exactly - as I understand it there is no use case for rendering case classes as a string with PII in-tact

I've overriden the `toString` behaviour for the remaining PII-containing case classes

Reply:
Thanks! 
Once you override the `toString` in case classes containing PII you won't need anymore to override `toString`s in other classes (like in the `Journey` cc).

Reply:
Looks like `toString` still needs to be overriden on more parent classes like `Journey` because it directly owns PII of its own (`ddRef` and `maybeSaUtr`)

But you'd still want `toString` behaviour overriden throughout in case you happen to be working with one of the more nested case classes directly

The behaviour I've introduced is that any case classes that define `obfuscate` also override `toString` and output the masked representation